include/libcyphal/transport/can/delegate.hpp

```diff
@@ -6,14 +6,21 @@
 #ifndef LIBCYPHAL_TRANSPORT_CAN_DELEGATE_HPP_INCLUDED
 #define LIBCYPHAL_TRANSPORT_CAN_DELEGATE_HPP_INCLUDED
 
-#include <libcyphal/transport/types.hpp>
-#include <libcyphal/transport/errors.hpp>
-#include <libcyphal/transport/scattered_buffer.hpp>
+#include "libcyphal/transport/errors.hpp"
+#include "libcyphal/transport/scattered_buffer.hpp"
+#include "libcyphal/transport/types.hpp"
+#include "libcyphal/types.hpp"
 
 #include <canard.h>
+#include <cetl/cetl.hpp>
+#include <cetl/pf17/attribute.hpp>
+#include <cetl/pf17/cetlpf.hpp>
 #include <cetl/rtti.hpp>
```

Comment:
nit: These are not technically system headers, so should we be using `""` instead of `<>`? We have been discussing this internally as well. And are leaning towards always using `""`. I know it's common to use `<>` for any headers that aren't part of the local repo, but it tells the compiler to first search the system headers and it won't find these headers there.

Reply:
These are all done like this intentionally and I believe quite consistently - everything which is not directly part of libcyphal is treated as "system" (or external). In unit tests I'm going even further - libcyphal itself is treated there like "external" stuff (kinda emulating how clients of our library supposed to use it). Also, AFAIK, `""` have "relative path notation" which is not applicable for any external stuff (whether it's a real system header or just dependency to a submodule like CETL).

Reply:
I am with Sergei here because per the standard, the behavior of `""` is a superset of `<>` -- if the former fails, it falls back to `<>`; if we know that it is going to fail there is no utility in using it.

<img width="629" alt="image" src="https://github.com/OpenCyphal-Garage/libcyphal/assets/3298404/fc360c6d-4837-4170-acaf-ebafe2078f3c">

Reply:
Thank you! I am aligned with continuing to use `<>` for "external" stuff.

@thirtytwobits FYI since we were just discussing this the other day.